ext/mbstring/mbstring.c

```diff
@@ -1715,13 +1715,85 @@ PHP_FUNCTION(mb_str_split)
 	}
 }
 
+#ifdef __SSE2__
```

Comment:
This has the drawback that binaries may not be portable, but I assume that `__SSE2__` is practically available everywhere (and we rely on that elsewhere), it shouldn't be an issue.

Reply:
True. SSE2 was introduced in the year 2000, and at this point, I think it would be pretty hard to find a x86-64 CPU which *doesn't* support it. (Please correct me if I'm wrong.) Actually, I think assuming *only* SSE2 support is very conservative at this point in time; just about all x86-64 CPUs support other extensions newer than that.

As you say, other parts of the PHP codebase do use SSE2 to accelerate various operations, like `strtoupper`.

Reply:
> I think it would be pretty hard to find a x86-64 CPU which doesn't support it. 

Correct. Every x86-64 CPU *must* implement SSE2 as per AMD64 spec, so you will not find a x86-64 CPU that doesn't support it.

Reply:
@nielsdos Thanks for explaining that.

Reply:
@nielsdos Thanks for explaining that.